src/components/chat/chat-comment-thread.ts

```diff
@@ -0,0 +1,40 @@
+import { WithId } from "src/hooks/firestore-hooks";
+import { CommentDocument } from "src/lib/firestore-schema";
+import { DocumentContentModelType } from "src/models/document/document-content";
+
+export interface ChatCommentThread {
+    title: string | null | undefined;
+    tileId: string | null;
+    tileType: string | null;
+    comments: WithId<CommentDocument>[];
+  }
+
+  // Expects a list of comments in Document order with document level comments (with no (null) tileId) first.
+  export function makeChatThreads (
+    commentsInDocumentOrder:WithId<CommentDocument>[] | undefined, 
+    content: DocumentContentModelType | undefined) : ChatCommentThread[] {
+    const chatThreads: ChatCommentThread[] = [];
+    // Since the comments are sorted by tile id already, each time we encounter a new tile id, we make a new thread.
+    commentsInDocumentOrder?.forEach((comment, idx)=> {
```

Comment:
I notice `idx` is not used yet - is it going to be for ordering the comments? They do come out in order for me at the moment.

Reply:
Oh whoops. No. I was using idx for the key, but switched to using the tile id instead and forgot to remove this. Removed!